core/templates/dev/head/pages/header_js_libs.html

```diff
@@ -1,8 +1,6 @@
 <script src="{{ASSET_DIR_PREFIX}}/assets/rich_text_components_definitions.js"></script>
 <script src="{{ASSET_DIR_PREFIX}}/assets/constants.js"></script>
-{% if not DEV_MODE %}
-  <script src="{{ASSET_DIR_PREFIX}}/assets/hashes.js"></script>
-{% endif %}
+<script ng-if="!DEV_MODE" src="{{ASSET_DIR_PREFIX}}/assets/hashes.js"></script>
```

Comment:
The problem is that Angular is loaded after this file. So this will be downloaded anyway.

Reply:
I am working on this issue and also got the same problem here. I couldn't use the Angular directive `ng-if` since the library itself has not been downloaded yet at that line. Any suggestion to solve this?

Reply:
The `hashes.js` specifies hashes for prod mode, we might get rid of this condition altogether and in dev mode load empty version of `hashes.js`.  `hashes.js` is generated in [scripts/build.py](https://github.com/oppia/oppia/blob/develop/scripts/build.py#L1057). The solution to this might be to create just an empty `hashes.js` file in assets
folder and ensure that in the build proces it will get replaced by a file with actual content.

Reply:
thanks for your solution